scipy/stats/_stats_py.py

```diff
@@ -7061,10 +7065,29 @@ def ttest_rel(a, b, axis=0, nan_policy='propagate', alternative="two-sided"):
 
     Returns
     -------
-    statistic : float or array
-        t-statistic.
-    pvalue : float or array
-        The p-value.
+    result : `~scipy.stats._result_classes.TtestResult`
+        An object with the following attributes:
+
+        statistic : float or array
+            The t-statistic.
+        pvalue : float or array
+            The p-value associated with the given alternative.
+        df : float or array
+            The number of degrees of freedom used in calculation of the
+            t-statistic; this is one less than the size of the sample
+            (``a.shape[axis]``).
+
+            .. versionadded:: 1.10.0
+
+        The object also has the following method:
+
+        confidence_interval(confidence_level=0.95)
```

Comment:
Not sure how to do this the best, but atm there is no space in the rendered doc. Very minor inconvenience.

Reply:
We're talking about this, right?
![image](https://user-images.githubusercontent.com/6570539/186971622-96a99986-5b90-4d23-8358-7400de6a5b03.png)

I'm not sure how to improve that either. It's not just that there is not much space space between this description "The object also has the following method" and the method name `confidence_interval`, but there is also the same, small amount of space between the description of the result object "An object with the following attributes." and the first attribute `statistic`.

![image](https://user-images.githubusercontent.com/6570539/186971645-188b6ea2-1106-436e-bda2-ea79bfcb7e8e.png)

It is the same in the documentation of all functions that are said to a return object.

![image](https://user-images.githubusercontent.com/6570539/186971724-1b786de9-4ef2-406f-86bf-ed0ff06937cb.png)

![image](https://user-images.githubusercontent.com/6570539/186971757-da897268-5764-4e18-93d3-eb21d5a59978.png)

![image](https://user-images.githubusercontent.com/6570539/186972001-2f189dfd-0499-4d84-860e-9fcc3e300482.png)

And not just in stats:
![image](https://user-images.githubusercontent.com/6570539/186972401-df994290-01c2-4588-bda4-9913ba98ab46.png)

So it seems like something to adjust across the board, if we want to do that.

Reply:
> I was afraid of backward compatibility at first 

That is the nice thing about `make_tuple_bunch`. It allows us to return objects with additional attributes and methods, while the original attributes still unpack like a regular tuple.

Reply:
Yes don't worry I am not asking for a fix here.

For the backward compatibility I was more referring to the name of the object. But this was added in the previous PR. So seems all good.

Reply:
I see. These result objects are not "public" in the sense that they cannot be imported from `stats`, so nobody should be relying on the name of them (e.g. using `isinstance`). (Of course, there's nothing to stop people from checking `str(type(res))`, but that's why it's not good style to do that... these things should be allowed to change.)

Reply:
Yes should not be an issue if this was never part of the public API (even if some persist and call private things). Otherwise some could do things like `isinstance`. Albeit there is limited usage of actually doing that in this case.